### PR TITLE
ddl: fix a panic when add column with insert an index column

### PR DESCRIPTION
**Pull request by jackysp**
PTAL @coocood @zimulala  back trace list below.
```
 2018/02/12 19:52:33.466 ddl_worker.go:51: [error] ddlWorker runtime error: index out of range goroutine 133 [running]:
github.com/pingcap/tidb/util.GetStack(0xc42038ef98, 0x1d08ac0, 0x28bd4d0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/util/misc.go:52 +0x74
github.com/pingcap/tidb/ddl.(*ddl).onDDLWorker.func1()
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl_worker.go:50 +0x6e
panic(0x1d08ac0, 0x28bd4d0)
	/usr/local/Cellar/go/1.9.3/libexec/src/runtime/panic.go:491 +0x283
github.com/pingcap/tidb/expression.TableInfo2SchemaWithDBName(0x1eb617c, 0xd, 0x1eb617c, 0xd, 0xc420a86700, 0xc4200c64d0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/expression/expression.go:274 +0x8fe
github.com/pingcap/tidb/plan.(*planBuilder).buildInsert(0xc420170300, 0xc420a8e1e0, 0x0, 0x1e2ff00)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/plan/planbuilder.go:765 +0xbf
github.com/pingcap/tidb/plan.(*planBuilder).build(0xc420170300, 0x26a1aa0, 0xc420a8e1e0, 0x0, 0xc420d3f950)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/plan/planbuilder.go:188 +0x26e
github.com/pingcap/tidb/plan.Optimize(0x26c8380, 0xc421390dd0, 0x26a1aa0, 0xc420a8e1e0, 0x26b3c60, 0xc420d9d8c0, 0xc42038fa00, 0x0, 0x0, 0x0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/plan/optimizer.go:68 +0x11e
github.com/pingcap/tidb/executor.(*Compiler).Compile(0xc42038fc88, 0x3972578, 0xc420086048, 0x26aa8a0, 0xc420a8e1e0, 0x0, 0x0, 0x0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/executor/compiler.go:48 +0x1eb
github.com/pingcap/tidb.(*session).Execute(0xc421390dd0, 0x3972578, 0xc420086048, 0x1ee7134, 0x38, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/session.go:805 +0xad4
github.com/pingcap/tidb/ddl_test.(*testStateChangeSuite).runTestInWriteOnly.func1(0xc420995340)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl_db_change_test.go:331 +0x171
github.com/pingcap/tidb/ddl.(*TestDDLCallback).OnJobUpdated(0xc4213993b0, 0xc420995340)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/callback_test.go:43 +0x64
github.com/pingcap/tidb/ddl.(*ddl).handleDDLJobQueue(0xc420a86000, 0x1, 0x0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl_worker.go:252 +0x159
github.com/pingcap/tidb/ddl.(*ddl).onDDLWorker(0xc420a86000)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl_worker.go:64 +0x224
created by github.com/pingcap/tidb/ddl.(*ddl).start
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl.go:318 +0x13a
```

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -32,6 +32,17 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+func (d *ddl) adjustTableColOffset(tblInfo *model.TableInfo, offset int) {
+	oldCols := tblInfo.Columns
+	newCols := make([]*model.ColumnInfo, 0, len(oldCols))
+	newCols = append(newCols, oldCols[:offset]...)
+	newCols = append(newCols, oldCols[len(oldCols)-1])
```
Comment (winkyao) [inline ddl/column.go]: add comments here.

**Comment by jackysp** [comment]
PTAL @winkyao

**Comment by coocood** [comment]
LGTM

**Pull request merged**